apis/changelogs/proto/v1alpha1/changelog.proto

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2024 The Crossplane Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+syntax = "proto3";
+
+import "google/protobuf/struct.proto";
+
+// buf:lint:ignore PACKAGE_DIRECTORY_MATCH
+package changelogs.proto.v1alpha1;
+
+option go_package = "github.com/crossplane/crossplane-runtime/apis/changelogs/proto/v1alpha1";
+
+// ChangeLogService is a service that provides the ability to send change log
+// entries.
+service ChangeLogService {
+  // SendChangeLog sends a change log entry to the change log service.
+  rpc SendChangeLog (SendChangeLogRequest) returns (SendChangeLogResponse) {}
+}
+
+// SendChangeLogRequest represents a request to send a single change log entry.
+message SendChangeLogRequest {
+  // The change log entry to send as part of this request.
+  ChangeLogEntry entry = 1;
+}
+
+// ChangeLog represents a single change log entry, with detailed information
+// about the resource that was changed.
+message ChangeLogEntry {
+  // The name and version of the provider that is making the change to the
+  // resource.
+  string provider = 1;
+
+  // The type of the resource that was changed, e.g. Group, Version, Kind.
+  string type = 2;
```

Comment:
Would it make sense to have distinct `apiVersion` and `kind` fields? This might make formatting clearer.

Reply:
You know, I've been recently thinking more and more of doing that myself. The current `type` field looks like this in the change logs:
```
"type":"kubernetes.crossplane.io/v1alpha2, Kind=Object"
```

I can see it being better for formatting, but also maybe better for indexing/searching to separate those into something like:
```
"apiVersion":"kubernetes.crossplane.io/v1alpha2","kind":"Object"
```

Reply:
Done in https://github.com/crossplane/crossplane-runtime/pull/754/commits/874a4a4f39c560a7ea64648d3154807e665857ec